From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sun, 22 Feb 2026 16:20:58 -0600
Subject: [PATCH 22/46] 
 0027-server-Check-scancode-to-distinguish-VK_RSHIFT-VK_LS.patch

---
 server/queue.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/queue.c b/server/queue.c
index 2815ec29b17..95cc2fcb1e3 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -2389,7 +2389,7 @@ static int queue_keyboard_message( struct desktop *desktop, user_handle_t win, c
     case VK_SHIFT:
     case VK_LSHIFT:
     case VK_RSHIFT:
-        vkey = (input->kbd.flags & KEYEVENTF_EXTENDEDKEY) ? VK_RSHIFT : VK_LSHIFT;
+        vkey = (input->kbd.scan == 0x36) ? VK_RSHIFT : VK_LSHIFT;
         if ((input->kbd.vkey & 0xff) == VK_SHIFT) hook_vkey = vkey;
         break;
     }
